addons/mail/models/mail_message.py

```diff
@@ -1007,11 +1007,22 @@ def _message_format(self, fnames):
                     })
 
             if message_sudo.model and message_sudo.res_id:
-                record_name = self.env[message_sudo.model] \
+                record = self.env[message_sudo.model] \
                     .browse(message_sudo.res_id) \
                     .sudo() \
-                    .with_prefetch(thread_ids_by_model_name[message_sudo.model]) \
-                    .display_name
+                    .with_prefetch(thread_ids_by_model_name[message_sudo.model])
+                record_name = record.display_name
+
+                if message_sudo.notification_ids.filtered(lambda n: n.notification_type == 'inbox'):
```

Comment:
Hello, thank you for the changes.

While this looks better in terms of query in general, this is not exactly what I was talking about in terms "only inbox".

I was talking about the inbox specific message fetch (and not whether the message had at least one inbox notification as done currently). This is still fetching and returning this information in cases where we don't need it.

I'm thinking `message_format`/`_message_format` should support option params, and in particular I would have an option that says if follower should be returned or not, and it would only be set to True when called from the message_fetch of inbox.